tests/components/zha/test_update.py

```diff
@@ -0,0 +1,585 @@
+"""Test ZHA firmware updates."""
+from unittest.mock import AsyncMock, MagicMock, call, patch
+
+import pytest
+from zigpy.exceptions import DeliveryError
+from zigpy.ota import CachedImage
+import zigpy.ota.image as firmware
+import zigpy.profiles.zha as zha
+import zigpy.types as t
+import zigpy.zcl.clusters.general as general
+import zigpy.zcl.foundation as foundation
+
+from homeassistant.components.homeassistant import (
+    DOMAIN as HA_DOMAIN,
+    SERVICE_UPDATE_ENTITY,
+)
+from homeassistant.components.update import (
+    ATTR_IN_PROGRESS,
+    ATTR_INSTALLED_VERSION,
+    ATTR_LATEST_VERSION,
+    DOMAIN as UPDATE_DOMAIN,
+    SERVICE_INSTALL,
+)
+from homeassistant.components.update.const import ATTR_SKIPPED_VERSION
+from homeassistant.const import ATTR_ENTITY_ID, STATE_OFF, STATE_ON, Platform
+from homeassistant.core import HomeAssistant, State
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.setup import async_setup_component
+
+from .common import async_enable_traffic, find_entity_id, update_attribute_cache
+from .conftest import SIG_EP_INPUT, SIG_EP_OUTPUT, SIG_EP_TYPE
+
+from tests.common import mock_restore_cache_with_extra_data
+
+
+@pytest.fixture(autouse=True)
+def update_platform_only():
+    """Only set up the update and required base platforms to speed up tests."""
+    with patch(
+        "homeassistant.components.zha.PLATFORMS",
+        (
+            Platform.UPDATE,
+            Platform.SENSOR,
+            Platform.SELECT,
+            Platform.SWITCH,
+        ),
+    ):
+        yield
+
+
+@pytest.fixture
+def zigpy_device(zigpy_device_mock):
+    """Device tracker zigpy device."""
+    endpoints = {
+        1: {
+            SIG_EP_INPUT: [general.Basic.cluster_id, general.OnOff.cluster_id],
+            SIG_EP_OUTPUT: [general.Ota.cluster_id],
+            SIG_EP_TYPE: zha.DeviceType.ON_OFF_SWITCH,
+        }
+    }
+    return zigpy_device_mock(
+        endpoints, node_descriptor=b"\x02@\x84_\x11\x7fd\x00\x00,d\x00\x00"
+    )
+
+
+async def setup_test_data(
+    zha_device_joined_restored,
+    zigpy_device,
+    skip_attribute_plugs=False,
+    file_not_found=False,
+):
+    """Set up test data for the tests."""
+    fw_version = 0x12345678
+    installed_fw_version = fw_version - 10
+    cluster = zigpy_device.endpoints.get(1).out_clusters[general.Ota.cluster_id]
+    if not skip_attribute_plugs:
+        cluster.PLUGGED_ATTR_READS = {"current_file_version": installed_fw_version}
```

Comment:
wanna use `general.Ota.AttributeDefs.current_file_version.name` here too?
(and other `AttributeDefs` in the test parts below, mostly in `make_packet`)